From b1062bcc262ce9f8658955d0d450280d2571d35d Mon Sep 17 00:00:00 2001
From: Huang Jiachai <hjc@rock-chips.com>
Date: Mon, 9 May 2016 09:30:58 +0800
Subject: [PATCH] video: rockchip: fb: don't set vop win state when switch
 screen

because hwc will open fb0 to fb5, so the logic state is enable,
this will lead to iommu page fault when switch screen.

Change-Id: I8bc34887a62338049d1d526e37a2595122265046
Signed-off-by: Huang Jiachai <hjc@rock-chips.com>
---
 drivers/video/rockchip/rk_fb.c | 15 ++++++++-------
 1 file changed, 8 insertions(+), 7 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index e7520eef07a1..a67cf66dbf5d 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -3691,13 +3691,14 @@ int rk_fb_switch_screen(struct rk_screen *screen, int enable, int lcdc_id)
 				} else if (!dev_drv->win[win_id]->state) {
 					dev_drv->ops->open(dev_drv, win_id, 1);
 					dev_drv->suspend_flag = 0;
-					mutex_lock(&dev_drv->win_config);
-					info->var.xoffset = 0;
-					info->var.yoffset = 0;
-					info->fbops->fb_set_par(info);
-					info->fbops->fb_pan_display(&info->var,
-								    info);
-					mutex_unlock(&dev_drv->win_config);
+					/* mutex_lock(&dev_drv->win_config);
+					 * info->var.xoffset = 0;
+					 * info->var.yoffset = 0;
+					 * info->fbops->fb_set_par(info);
+					 * info->fbops->fb_pan_display(&info->var,
+					 *			    info);
+					 * mutex_unlock(&dev_drv->win_config);
+					 */
 				}
 			}
 		}
-- 
2.35.3

